pkg/nsx/services/inventory/compare.go

```diff
@@ -17,19 +17,17 @@ func compareResources(pre interface{}, cur interface{}) map[string]interface{} {
 	case string(ContainerApplicationInstance):
 		compareContainerApplicationInstance(pre, cur, &updateProperties)
 	}
+	log.Info("Compare resource", "updateProperties", updateProperties)
 	return updateProperties
 }
 
 func compareContainerProject(pre interface{}, cur interface{}, property *map[string]interface{}) {
 	updateProperties := *property
-	if pre == nil {
+	if pre == nil || !reflect.DeepEqual(pre.(containerinventory.ContainerProject).Tags, cur.(containerinventory.ContainerProject).Tags) {
 		updateProperties["display_name"] = cur.(containerinventory.ContainerProject).DisplayName
```

Comment:
the pre == nil here mean no record found in the db.
in your case, it seems pre doesn't work if you not update the code in build.go.
and you still need to add:
updateProperties["tags"] = cur.(containerinventory.ContainerProject).Tags

Reply:
fixed